iset.mm

```diff
@@ -39258,6 +39275,30 @@ rather than an implication (but the two are equivalent by ~ bm1.3ii ),
       VHABVFBJZVTVHWDVTTVFVGRZVGVFRZTVHWDWEVTWFVFBUQVGVFURVAVFVGUSUTVBVCVDVE $.
   $}
 
+  $( Existence of a conditional set, quantitative version (closed form).
+     (Contributed by BJ, 15-Aug-2024.) $)
+  ifexg $p |- ( ( A e. V /\ B e. W ) -> if ( ph , A , B ) e. ~P ( A u. B ) ) $=
```

Comment:
I wouldn't call these `ifexg`, `ifexd`, and `ifex`, both because of "Theorems which are different [between set.mm and iset.mm] should be named differently" from https://us.metamath.org/ileuni/conventions.html but also because it is a pretty well established convention in both set.mm and iset.mm that `ex` means set existence, not some flavor of closure (or whatever the word is for `e. ~P ( A u. B )` as opposed to `e. _V`).

I realize that you wanted the stronger theorem to prove `fmelpw1o`, which does seem like a noble goal, but a name like `ifpw` or `ifelpw` would seem fine. Also, given the limited usage of these, I think we probably just need one (the `g` or `d` one, I don't have much of an opinion between those two) rather than three.

Separately, it would be fine to have `ifex` and `ifexd` stated exactly as in set.mm. As far as I could tell from git logs, I didn't have any reason for leaving them out other than that the set.mm proof doesn't work as-is and I didn't have a need for them (well, that I noticed - as you pointed out in the description of this pull request there might be places they could shorten some existing proofs).

Reply:
Ok, I'm going to change the labels. Do you have some proposals ? Maybe ifsspw(|d|g) ?

Then, I'll prove the standard form of ifex(|d|g) from them.

I had chosen the same labels, because to me, this was almost the same as existence: when existence is needed, it is often of the for `... e. V ` where ` V ` is a metavariable, so any version of existence would do.

"Quantitative" versions are  nice because they permit to work inside various kinds of universes: if you have a universe closed under unions and powersets, then you have existence in that universe.

Reply:
Done. I chose to add the deduction version only, but I think the other versions (closed form and inference form) could have been added too. In general, I think we should not shy away from adding these forms: currently, the more limited library of lemmas in iset.mm makes it harder (at least for me) to prove things than in set.mm, because the proof assistant "matches" fewer possibilities (I mean: the `IMPROVE ALL` of `MM-PA` or the `Ctrl+U` of `mmj2` are less likely to finish the proof for you).

Reply:
There isn't any intentional divergence between iset.mm and set.mm on this point. If it is a frequently used theorem, it is fine to have multiple forms.